src/Database/LSMTree/Internal.hs

```diff
@@ -1,5 +1,12 @@
 {-# LANGUAGE DataKinds #-}
 
+-- | The internal API here is defined in terms of untyped serialised keys,
+-- values and blobs. It makes no distinction between normal and monoidal tables,
+-- accepting both blobs and mupserts.
+--
+-- This module mainly deals with concurrency- and exception-safe opening and
+-- closing of resources. Any other non-trivial logic should live somewhere else.
+--
```

Comment:
Nice, it is indeed helpful if we record the intention of the module here. Maybe we should also note here that this module is, in a way, bringing all internal parts together to implement the public API.